client/components/category-navigation/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { Button } from '@wordpress/components';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import classnames from 'classnames';
+
+import './style.scss';
+
+type CategoryNavProps = {
+	buttons?: {
+		icon: string;
+		label: string;
+		link: string;
+	}[];
+	list: {
+		name: string;
+		label?: string;
+		link: string;
+	}[];
+	selectedCategory: string;
+};
+
+export const CategoryNavigation = ( { buttons, list, selectedCategory }: CategoryNavProps ) => {
+	const [ showLeftArrow, setShowLeftArrow ] = useState( false );
+	const [ showRightArrow, setShowRightArrow ] = useState( false );
+	const listRef = useRef< HTMLDivElement | null >( null );
+
+	const checkScrollArrows = () => {
+		const list = listRef.current;
+		if ( ! list ) {
+			return;
+		}
+
+		const { scrollLeft, scrollWidth, clientWidth } = list;
```

Comment:
```suggestion
		if ( ! listRef.current ) {
			return;
		}

		const { scrollLeft, scrollWidth, clientWidth } = listRef.current;
```
Nit, but I see no reason to abstract `listRef.current` into a temporary variable.